File: app/src/main/java/com/worldventures/dreamtrips/modules/config/service/storage/UpdateRequirementStorage.java
package com.worldventures.dreamtrips.modules.config.service.storage;

import android.support.annotation.Nullable;

import com.worldventures.core.janet.cache.CacheBundle;
import com.worldventures.core.janet.cache.CachedAction;
import com.worldventures.core.janet.cache.storage.ActionStorage;
import com.worldventures.dreamtrips.core.repository.SnappyRepository;
import com.worldventures.dreamtrips.modules.config.model.Configuration;
import com.worldventures.dreamtrips.modules.config.service.command.ConfigurationCommand;

public class UpdateRequirementStorage implements ActionStorage<Configuration> {

   private SnappyRepository snappyRepository;

   public UpdateRequirementStorage(SnappyRepository snappyRepository) {
      this.snappyRepository = snappyRepository;
   }

   @Override
   public Class<? extends CachedAction> getActionClass() {
      return ConfigurationCommand.class;
   }

   @Override
   public void save(@Nullable CacheBundle params, Configuration data) {
      snappyRepository.saveAppUpdateRequirement(data);
   }

   @Override
   public Configuration get(@Nullable CacheBundle action) {
      return snappyRepository.getAppUpdateRequirement();
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/config/service/storage/UpdateRequirementStorage.kt
package com.worldventures.dreamtrips.modules.config.service.storage

import com.worldventures.core.janet.cache.CacheBundle
import com.worldventures.core.janet.cache.storage.ActionStorage
import com.worldventures.dreamtrips.core.repository.SnappyRepository
import com.worldventures.dreamtrips.modules.config.model.Configuration
import com.worldventures.dreamtrips.modules.config.service.command.ConfigurationCommand

class UpdateRequirementStorage(private val snappyRepository: SnappyRepository) : ActionStorage<Configuration> {

   override fun getActionClass() = ConfigurationCommand::class.java

   override fun save(params: CacheBundle?, data: Configuration) = snappyRepository.saveAppUpdateRequirement(data)

   override fun get(action: CacheBundle?) = snappyRepository.appUpdateRequirement
}
